client/test/utils/mockery_utils.js

```diff
@@ -0,0 +1,38 @@
+const mockery = require('mockery');
+
+
+// Bunch of functions to make tests less verbosed.
+module.exports = {
+
+  // Register test dispatcher as a replacement of the official dispatcher.
+  initDispatcher: (dispatcher) => {
+    mockery.registerMock('../dispatcher/dispatcher', dispatcher);
+    mockery.registerMock(
+      '../libs/flux/dispatcher/dispatcher', dispatcher);
```

Comment:
Because mockery register module path, not the module itself. That way, when you use a `require` it loads the mock by comparing require argument with the paths you registered.